packages/shared/src/urbit/channel.ts

```diff
@@ -357,6 +371,53 @@ export interface Perm {
   group: Flag;
 }
 
+interface Bot {
+  nickname?: string;
+  avatar?: string;
+  origin?: string;
+  agent: string;
+}
+
+interface PostMetadataSchemaV1 {
+  type: string;
+  bot?: Bot;
+  rendererOverride?: string;
+}
+
+interface MessageInput {
+  type: string;
+  postType: string;
+  configuration: Record<string, string>;
+}
+
+interface ContentRenderer {
+  render: string;
+  using: string;
+}
+
+interface ChannelMetadataSchemaV1 {
+  version: 1;
+  messageInputs: MessageInput[];
+  defaultContentRenderers?: ContentRenderer[];
+  defaultPostCollectionRenderer?: string;
+}
+
+type ChannelMetadataSchema = ChannelMetadataSchemaV1;
+
+interface Hook {
+  version: number;
+  name: string;
+  src: string;
+  compiled: boolean;
```

Comment:
what does this mean?

Reply:
I was thinking we might need some indicator of whether this source code actually compiles or not so we could surface an error that something is wrong with your code.

Reply:
seems useful - i read this as "is this code precompiled" - maybe "valid" would be more intuitive?